forester/src/main.rs

```diff
@@ -103,48 +107,120 @@ async fn main() {
+        let rollover_state = rollover_state.clone();
+        let handle = tokio::spawn(async move {
+            match tree_data.tree_type {
+                TreeType::State => {
+                    run_subscribe_state(
+                        config_clone.clone(),
+                        pool_clone.clone(),
+                        tree_data,
+                        rollover_state.clone(),
+                    )
+                    .await
+                }
+                TreeType::Address => {
+                    run_subscribe_addresses(
+                        config_clone.clone(),
+                        pool_clone.clone(),
+                        tree_data,
+                        rollover_state.clone(),
+                    )
+                    .await
+                }
+            }
+        });
+        handles.push(handle);
+    }
+
+    // Wait for all nullification processes to complete
+    for handle in handles {
+        if let Err(e) = handle.await {
+            error!("A nullification process encountered an error: {:?}", e);
+        }
+    }
+
+    debug!("All nullification processes completed");
+}
+
+async fn run_subscribe_state(
+    config: Arc<ForesterConfig>,
+    rpc_pool: RpcPool<SolanaRpcConnection>,
+    tree_data: TreeData,
+    rollover_state: Arc<RolloverState>,
+) {
     let indexer_rpc = init_rpc(config.clone(), false).await;
     let indexer = Arc::new(tokio::sync::Mutex::new(PhotonIndexer::new(
         config.external_services.indexer_url.to_string(),
         indexer_rpc,
     )));
 
-    subscribe_state(config.clone(), rpc_pool, indexer).await;
+    subscribe_state(config.clone(), rpc_pool, indexer, tree_data, rollover_state).await;
 }
 
 async fn run_subscribe_addresses<R: RpcConnection>(
     config: Arc<ForesterConfig>,
     rpc_pool: RpcPool<R>,
+    tree_data: TreeData,
+    rollover_state: Arc<RolloverState>,
 ) {
     let indexer_rpc = init_rpc(config.clone(), false).await;
     let indexer = Arc::new(tokio::sync::Mutex::new(PhotonIndexer::new(
         config.external_services.indexer_url.to_string(),
         indexer_rpc,
     )));
 
-    subscribe_addresses(config.clone(), rpc_pool, indexer).await;
+    subscribe_addresses(config.clone(), rpc_pool, indexer, tree_data, rollover_state).await;
 }
 
-async fn run_nullify_state<R: RpcConnection>(config: Arc<ForesterConfig>, rpc_pool: RpcPool<R>) {
+async fn run_nullify_addresses<R: RpcConnection>(
+    config: Arc<ForesterConfig>,
+    rpc_pool: RpcPool<R>,
+    tree_data: TreeData,
+    rollover_state: Arc<RolloverState>,
+) {
     let indexer_rpc = init_rpc(config.clone(), false).await;
     let indexer = Arc::new(tokio::sync::Mutex::new(PhotonIndexer::new(
         config.external_services.indexer_url.to_string(),
         indexer_rpc,
     )));
 
-    nullify_state(config.clone(), rpc_pool, indexer).await;
+    nullify_addresses(config.clone(), rpc_pool, indexer, tree_data, rollover_state).await;
 }
```

Comment:
Are these still used?
It looks like `run_subscribe` covers the same functionality unified right?